keps/sig-network/20190603-endpointslices/README.md

```diff
@@ -626,6 +678,173 @@ address types.
+  - endpoint_slice_controller/endpoints_desired
+	- endpoint_slice_controller/num_endpoint_slices
+	- endpoint_slice_controller/desired_endpoint_slices
+	- endpoint_slice_controller/changes
+
+* **What are the SLIs (Service Level Indicators) an operator can use to
+  determine the health of the service?**
+  - [x] Metrics
+    - Metric name: network_programming_duration_seconds
+    - Components exposing the metric: This requires cooperation between the
+      EndpointSlice controller and kube-proxy. The metric itself is exposed by
+      kube-proxy. This indicates how long it takes for proxy rules to be updated
+      after a Kubernetes resource changes.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  The network_programming_duration_seconds metric should be under 60 at least
+  95% of the time.
+
+* **Are there any missing metrics that would be useful to have to improve
+  observability if this feature?**
+  Since this is a networking feature, it would be helpful to have more metrics
+  covering things like error rates. Unfortunately request errors can be caused
+  by a wide variety of things so it would likely be a very noisy metric.
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+  Enabling the EndpointSliceProxying feature gate on kube-proxy is dependent
+  on the EndpointSlice feature gate being enabled on kube-controller-manager
+  and kube-apiserver. This feature does not depend on any additional services.
+
+### Scalability
+
+
+* **Will enabling / using this feature result in any new API calls?**
+  Enabling the EndpointSlice controller will result in significant numbers of
+  write (create, update, delete) requests to the EndpointSlice API. The
+  throughput will depend on the number of EndpointSlices required and the
+  frequency with which they'll be updated. 
+  
+  Each EndpointSlice can store up to 100 endpoints by default. If most Services
+  in a cluster have less than 100 endpoints, the load from the EndpointSlice
+  controller will be similar to the Endpoints controller. If many Services have
+  more than 100 endpoints, more EndpointSlices will be required and therefore
+  additional write requests will be sent to the API.
+  
+  The frequency that EndpointSlices are updated depends on how frequently
+  Services or their endpoints (typically Pods) are updated. Frequent Pod
+  replacement will result in frequent EndpointSlice updates - similar to the
+  behavior of the Endpoints controller.
+  
+  Enabling the EndpointSliceProxying feature gate for kube-proxy will result in
+  changes to read API calls. Instead of watching Endpoints resources, kube-proxy
+  will watch EndpointSlices. This should result in significant decreases in
+  bandwidth for large clusters with large Services. 
+  
+* **Will enabling / using this feature result in introducing new API types?**
+  Describe them providing:
+  - API type: EndpointSlice
```

Comment:
Please also add: `Supported number of objects per cluster`

Reply:
Good catch, added.